haskell/src/Monocle/Env.hs

```diff
@@ -111,6 +111,12 @@ type QueryM = ReaderT QueryEnv TenantM
 runQueryM :: Q.Query -> QueryM a -> TenantM a
 runQueryM query qm = runReaderT qm (QueryEnv query Nothing)
 
+-- | 'runQueryStream' run a QueryM stream into a TenantM
+-- Checkout the tutorial about mmoprh:
+--   https://hackage.haskell.org/package/mmorph-1.2.0/docs/Control-Monad-Morph.html#g:3
+runQueryStream :: Q.Query -> Stream (Of elem) QueryM res -> Stream (Of elem) TenantM res
+runQueryStream query = hoist (runQueryM query)
```

Comment:
Could this has been implemented w/o the use of a new package ?

Reply:
This is not a new package, mmorph is a dependency of streaming, see the MFunctor instance of https://hackage.haskell.org/package/streaming-0.2.3.0/docs/Streaming-Internal.html#t:Stream . If I understand correctly, `hoist` is the name of facility to operate on embedded context (e.g. when the m is not a prefix, as in `Parent m res` instead of the usual `m res`). We are doing the same thing to replace the Handler with our AppM when starting the servant server with https://hackage.haskell.org/package/servant-server-0.18.3/docs/Servant-Server.html#v:hoistServer , see https://github.com/change-metrics/monocle/blob/master/haskell/src/Monocle/Api.hs#L25.